### ISSUE TITLE
Loss near 0 but only 50% accuracy on training set

### ISSUE THREAD

**Issue by VisionEp1** [davisking/dlib#1893]
Hi,
im am currently very confused.
during training i get very low loss:
step#: 99209  learning rate: 0.001  train loss: 0.00171392

but when i test network on the try data it produces 
val top2 accuracy:  0.528
val top1 accuracy:  0.376

shouldn't the accuracy be around 100% with a loss very close to 0 ?
i checked the data which gets used as input and its all correct (image_win + cout during training loop)

Code for testing:

> for (auto l : list) {

			try {
			dlib::array<matrix<rgb_pixel>> images;
			matrix<rgb_pixel> img;
			load_image(img, l.filename);
				if(first)
				{
					first = false;
					cout << l.filename << endl;
					}	
			const int num_crops = 8;
			randomly_crop_images(img, images, rnd, num_crops);
			// p(i) == the probability the image contains object of class i.
			matrix<float, 1, numClasses> p = sum_rows(mat(snet(images.begin(), images.end()))) / num_crops;

			// check top 1 accuracy
			if (index_of_max(p) == l.numeric_label)
				++num_right_top1;
			else
				++num_wrong_top1;

			// check top 5 accuracy
			bool found_match = false;
			for (int k = 0; k < 2; ++k)
			{
				long predicted_label = index_of_max(p);
				p(predicted_label) = 0;
				if (predicted_label == l.numeric_label)
				{
					found_match = true;
					break;
				}

			}
			if (found_match)
				++num_right;
			else
				++num_wrong;
		}
		catch (...) {}
		}

		
		cout << "val top2 accuracy:  " << num_right / (double)(num_right + num_wrong) << endl;
		cout << "val top1 accuracy:  " << num_right_top1 / (double)(num_right_top1 + num_wrong_top1) << endl;
	}

**Comment by VisionEp1** [comment]
stupid me. i didnt use affine transformation instead of bn during testing
